src/queries/unsafeProps.ts

```diff
@@ -0,0 +1,103 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import prettyFormat from 'pretty-format';
+import { ErrorWithStack, prepareErrorMessage } from '../helpers/errors';
+import {
+  createQueryByError,
+  throwRemovedFunctionError,
+} from '../helpers/errors';
+
+const UNSAFE_getByProps = (
+  instance: ReactTestInstance
+): ((props: { [propName: string]: any }) => ReactTestInstance) =>
+  function getByPropsFn(props: { [propName: string]: any }) {
+    try {
+      return instance.findByProps(props);
+    } catch (error) {
+      throw new ErrorWithStack(prepareErrorMessage(error), getByPropsFn);
+    }
+  };
+
+const UNSAFE_getAllByProps = (
+  instance: ReactTestInstance
+): ((props: { [propName: string]: any }) => Array<ReactTestInstance>) =>
+  function getAllByPropsFn(props: { [propName: string]: any }) {
+    const results = instance.findAllByProps(props);
+    if (results.length === 0) {
+      throw new ErrorWithStack(
+        `No instances found with props:\n${prettyFormat(props)}`,
+        getAllByPropsFn
+      );
+    }
+    return results;
+  };
+
+const UNSAFE_queryByProps = (
+  instance: ReactTestInstance
+): ((props: { [propName: string]: any }) => ReactTestInstance | null) =>
+  function queryByPropsFn(props: { [propName: string]: any }) {
+    try {
+      return UNSAFE_getByProps(instance)(props);
+    } catch (error) {
+      return createQueryByError(error, queryByPropsFn);
+    }
+  };
+
+const UNSAFE_queryAllByProps = (
+  instance: ReactTestInstance
+): ((props: {
+  [propName: string]: any;
+}) => Array<ReactTestInstance>) => (props: { [propName: string]: any }) => {
+  try {
+    return UNSAFE_getAllByProps(instance)(props);
+  } catch (error) {
+    return [];
+  }
+};
+
+// Unsafe aliases
+export type UnsafeByPropsQueries = {
+  UNSAFE_getByProps: (props: { [key: string]: any }) => ReactTestInstance;
+  UNSAFE_getAllByProps: (props: {
+    [key: string]: any;
+  }) => Array<ReactTestInstance>;
+  UNSAFE_queryByProps: (props: {
+    [key: string]: any;
+  }) => ReactTestInstance | null;
+  UNSAFE_queryAllByProps: (props: {
+    [key: string]: any;
+  }) => Array<ReactTestInstance>;
+
+  // Removed aliases
+  // TODO: remove in the next release
+  queryByName: () => void;
```

Comment:
this should be in the unsafeByName file I guess (same for the other mentions of byName in the rest of the file)

Reply:
This is unmodified name from before the PR and it's there for easier code migration for users.

That being said this code deprecation is quite old now, all users should have migrated a long time ago, so I would be for removing these legacy aliases. @thymikee @AugustinLF wdyt about remove all these deprecated aliases?

Reply:
I've removed all removed functions error messages.